drivers/ieee802154/ieee802154_nrf5.c

```diff
@@ -433,42 +433,75 @@ static void nrf5_tx_started(const struct device *dev,
 	}
 }
 
+static bool nrf5_tx_immediate(struct net_pkt *pkt, uint8_t *payload, bool cca)
+{
+	if ((pkt->ieee802154_fc_is_set && (!pkt->ieee802154_ie_is_updated)) ||
```

Comment:
This is equivalent to: `if (pkt->ieee802154_fc_is_set  != pkt->ieee802154_ie_is_updated) {`

However it's probably being optimized out by the compiler. But for the current state of the radio driver APIs it can be moved safely from all the three transmission functions to `nrf5_tx` to reduce the code size.

Reply:
That's a very good catch, thank you.